vinhloc30796/creativecontact-app
# feat(profile)!: implement unified industry-experience UI

## Description

Restructures how we handle industry and experience data across the application, introducing a unified industry-experience model using combo badges for consistent UI representation.

## Key Changes

1. Introduced new `ComboBadge` component for consistent industry-experience display
2. Refactored professional info data structure to combine industry and experience levels
3. Updated form handling in ProfessionalSection and ProfileCard components
4. Standardized styling using primary color with opacity variants
5. Migrated existing industry/experience separate fields to combined model

## Breaking Changes

- Changed data structure from separate `industries` and `experience` fields to combined `industryExperiences` array
- Modified form schemas and validation logic for professional information
- Updated API response structure for user profile data

## Related Issues

- Closes WEB-214

## Testing Instructions

1. Navigate to profile editing page
2. Verify industry-experience combo selection works:
   - Add new industry with experience level
   - Remove existing combinations
   - Edit existing combinations
3. Check display consistency across:
   - Profile card
   - Contact cards
   - Professional section
4. Verify form validation and submission
5. Test responsive design of combo badges

## Additional Notes

- This change improves UI consistency and data organization
- The new combo badge component uses primary color with 80% opacity variant for visual hierarchy
- Migration scripts may be needed for existing user data
- Component changes affect multiple views but maintain backwards compatibility in the API layer
